gexpect_test.go

```diff
@@ -117,6 +117,7 @@ var regexMatchTests = []struct {
 	{`a+hello`, `aaaahello`, `bhello`},
 	{`(hello|world)`, `hello`, `unknown`},
 	{`(hello|world)`, `world`, `unknown`},
+	{"世", "世", `unknown`}, // 3 bytes long unicode character (from example of https://golang.org/pkg/unicode/utf8/#FullRune)
```

Comment:
This test is okeyish, because it tests that we return the utf8 runes correctly, but it actually does not test the fix you have made. This is because we would need to setup a buffer in such a way that `buf.b` has first two bytes of the three-byte utf8 character and `buf.f` - the rest of the character.